Makefile

```diff
@@ -0,0 +1,40 @@
+# define the name of the virtual environment directory
+VENV := venv
+
+# default target, when make executed without arguments
+all: venv
+
+install: FORCE
+	pip install --upgrade pip
+	pip install -r requirements.txt --require-venv
+	pip install -r requirements-optional.txt --require-venv
+	pip install --editable . --require-venv
+
+requirements.txt:
+	pip install -r requirements.txt --require-venv
+
+# update is a shortcut target
+update: requirements.txt
+
+test: update
+	# Executes pytests which are located in tests/
+	pytest
+	# Executes filecheck tests
+	lit tests/filecheck
+
+format:
+	yapf -ir xdsl
+
+lint:
+	pyright
+
+lint-ci: pyright-ci.json
```

Comment:
Why is `pyright-ci.json` a requirement here? Do we need to keep track of it?

Reply:
I guess it's not really useful to have it in the requirement slot